rules/aip0233/resource_reference_type.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0233
+
+import (
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+	"google.golang.org/genproto/googleapis/api/annotations"
+)
+
+// BatchCreate methods should reference the target resource via `child_type` or
+// the parent directly via `type`.
+var resourceReferenceType = &lint.MethodRule{
+	Name: lint.NewRuleName(233, "resource-reference-type"),
+	OnlyIf: func(m *desc.MethodDescriptor) bool {
+		out := m.GetOutputType()
+		if out.GetName() == "Operation" {
+			info := utils.GetOperationInfo(m)
+			out = utils.FindMessage(m.GetFile(), info.GetResponseType())
+		}
+
+		// First repeated message field must be annotated with google.api.resource.
+		repeated := utils.GetRepeatedMessageFields(out)
+		var resource *annotations.ResourceDescriptor
+		if len(repeated) > 0 {
```

Comment:
IIUC, if `len(repeated)` is zero then resource will alway be `nil` and the condition at the end of this statement will always be false. Hence, let's change this to,

```
if len(repeated) == 0 {
  return false
}
```

then, whole block can be outdented and `resource` can be declared with `:=` rather than `var`, etc.

Reply:
Done.